src/overworld.c

```diff
@@ -1,19 +1,3190 @@
+        (*state)++;
+        break;
+    case 3:
+        sub_80863B0();
+        map_loading_lcd_reset();
+        sub_8197200();
+        (*state)++;
+        break;
+    case 4:
+        move_tilemap_camera_to_upper_left_corner();
+        (*state)++;
+        break;
+    case 5:
+        copy_map_tileset1_to_vram(gMapHeader.mapData);
+        (*state)++;
+        break;
+    case 6:
+        copy_map_tileset2_to_vram(gMapHeader.mapData);
+        (*state)++;
+        break;
+    case 7:
+        if (free_temp_tile_data_buffers_if_possible() != TRUE)
+        {
+            apply_map_tileset1_tileset2_palette(gMapHeader.mapData);
+            (*state)++;
+        }
+        break;
+    case 8:
+        DrawWholeMapView();
+        (*state)++;
+        break;
+    case 9:
+        cur_mapheader_run_tileset_funcs_after_some_cpuset();
+        (*state)++;
+        break;
+    case 11:
+        if (gWirelessCommType != 0)
+        {
+            sub_800E0E8();
+            CreateWirelessStatusIndicatorSprite(0, 0);
+        }
+        (*state)++;
+        break;
+    case 12:
+        if (map_post_load_hook_exec())
+            (*state)++;
+        break;
+    case 10:
+        (*state)++;
+        break;
+    case 13:
+        SetFieldVBlankCallback();
+        (*state)++;
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void do_load_map_stuff_loop(u8 *state)
+{
+    while (!load_map_stuff(state, 0));
+}
+
+static void sub_80867C8(void)
+{
+    sub_81BE6AC();
+    MoveSaveBlocks_ResetHeap();
+}
+
+static void sub_80867D8(void)
+{
+    SetGpuReg(REG_OFFSET_DISPCNT, 0);
+    ScanlineEffect_Stop();
+
+    DmaClear16(3, PLTT + 2, PLTT_SIZE - 2);
+    DmaFillLarge16(3, 0, (void *)(VRAM + 0x0), 0x18000, 0x1000);
+    ResetOamRange(0, 128);
+    LoadOam();
+}
+
+static void sub_8086860(void)
+{
+    sub_80863B0();
+    map_loading_lcd_reset();
+    sub_8197200();
+    mapdata_load_assets_to_gpu_and_full_redraw();
+}
+
+static void map_loading_lcd_reset(void)
+{
+    clear_scheduled_bg_copies_to_vram();
+    reset_temp_tile_data_buffers();
+    SetGpuReg(REG_OFFSET_MOSAIC, 0);
+    SetGpuReg(REG_OFFSET_WININ, 0x1F1F);
+    SetGpuReg(REG_OFFSET_WINOUT, 0x101);
+    SetGpuReg(REG_OFFSET_WIN0H, 0xFF);
+    SetGpuReg(REG_OFFSET_WIN0V, 0xFF);
+    SetGpuReg(REG_OFFSET_WIN1H, 0xFFFF);
+    SetGpuReg(REG_OFFSET_WIN1V, 0xFFFF);
+    SetGpuReg(REG_OFFSET_BLDCNT, gUnknown_82EC7C4[1] | gUnknown_82EC7C4[2] | gUnknown_82EC7C4[3] | 0x1040);
+    SetGpuReg(REG_OFFSET_BLDALPHA, 0x70D);
+    overworld_bg_setup();
+    schedule_bg_copy_tilemap_to_vram(1);
+    schedule_bg_copy_tilemap_to_vram(2);
+    schedule_bg_copy_tilemap_to_vram(3);
+    ChangeBgX(0, 0, 0);
+    ChangeBgY(0, 0, 0);
+    ChangeBgX(1, 0, 0);
+    ChangeBgY(1, 0, 0);
+    ChangeBgX(2, 0, 0);
+    ChangeBgY(2, 0, 0);
+    ChangeBgX(3, 0, 0);
+    ChangeBgY(3, 0, 0);
+    SetGpuReg(REG_OFFSET_DISPCNT, DISPCNT_OBJ_ON | DISPCNT_WIN0_ON | DISPCNT_WIN1_ON | DISPCNT_OBJ_1D_MAP | 0x20);
```

Comment:
Add entry in io_reg.h, this is the data from GBATEK:
`Bit 5     H-Blank Interval Free  (1=Allow access to OAM during H-Blank)`